update.go

```diff
@@ -363,6 +363,10 @@ other options are ignored.
 				}
 				config.IntelRdt = &configs.IntelRdt{}
 				intelRdtManager := intelrdt.NewManager(&config, container.ID(), state.IntelRdtPath)
+				if intelRdtManager == nil {
+					logrus.Errorf("Intel RDT manager creation failed: likely due to missing resctrl filesystem")
+					return nil
```

Comment:
We should not return nil here. I think the entire logic should be changed like this:
```go
if intelRdtManager != nil && err := intelRdtManager.Apply(state.InitProcessPid); err != nil {
```

Reply:
Like my comment in https://github.com/opencontainers/runc/pull/4774#discussion_r2120888042
I think that one is more reasonable, WDYT?